### PR TITLE
Remove unnecessary lint rule suppression

### PR DESCRIPTION
**Pull request by PaulACoroneos**
Edit:

See updated comments from https://github.com/bluesky-social/social-app/pull/6544#issuecomment-2487273639. Thank you!

--- Original comments ---

Hello.

As part of the path to eslint v9 we need to [remove eslint-config-react-native-community which has not been updated in 2 years](https://www.npmjs.com/package/@react-native-community/eslint-config). This has [been replaced with react-native/eslint-plugin](https://www.npmjs.com/package/@react-native/eslint-plugin). As part of this migration eslint-plugin-react-hooks is no longer a dependency so I added it as well. (I also have added version detect so the react plugins know what version they are dealing with)

While doing this 2 rules-of-hooks violations were flagged. Both are valid as a hooks should not be called in an asynchronous function. That being said I don't really see a reason for the function to be marked asynchronous in first place. So I have remove the async from each declaration. This resolves the violation.

Finally we are able to formally remove references to prettier in the .eslintrc as the new plugins do not depend on eslint-plugin-prettier. Format concerns are now properly handled by prettier.

I ran yarn lint and validated we only get warnings back. Please let me know if you have any questions or need further details. Thank you!

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 module.exports = {
   root: true,
   extends: [
-    '@react-native-community',
     'plugin:react/recommended',
```
Comment (gaearon) [inline .eslintrc.js:4]: do we want `@react-native` here too? for the config.

Comment (gaearon) [reply .eslintrc.js:4]: nvm got it via https://github.com/bluesky-social/social-app/pull/6525

**Review by gaearon — APPROVED** [review]
let's try that

**Comment by gaearon** [comment]
needs a rebase pls

File: .eslintrc.js (hunk 1 above)
Comment (PaulACoroneos) [reply .eslintrc.js:4]: Haha beat me to it no worries :)

**Comment by PaulACoroneos** [comment]
@gaearon  rebase complete

File: .eslintrc.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
 module.exports = {
   root: true,
   extends: [
-    '@react-native',
```
Comment (trivikr) [inline .eslintrc.js:4]: Is this removal intentional, or happened by mistake during rebase? This is the new ESLint config.

File: .eslintrc.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
     'plugin:react/recommended',
+    'plugin:react-hooks/recommended',
     'plugin:react/jsx-runtime',
     'plugin:react-native-a11y/ios',
-    'prettier',
```
Comment (trivikr) [inline .eslintrc.js:8]: Please remove `eslint-config-prettier` dependency too from package.json

It's usually better to not mix multiple changes in one PR, but it's up to the maintainers to comment on it. The removal of `eslint-config-prettier` can be done independently of using `react-hooks`

Comment (PaulACoroneos) [reply .eslintrc.js:8]: Yes it is unnecessary. I am sorting through some stuff around prettier right now so I have not had a chance to update. The inclusion was because moving to plugin the rule wasn't being extended anymore. Thank you and kindly give me a bit to sort things out.

**Comment by PaulACoroneos** [comment]
@trivikr @gaearon Okay everything is fixed up (and I've now updated title after rebase) . Let me summarize:

I did not rebase tonight before starting work so I didn't see @trivikr beat me to the eslint package update. Thank you for doing that :).

When we made the update as @gaearon wanted we now relied on eslint-config-prettier which by configuration disables style rules such as semi, curly, etc. This is because we prefer to rely on the prettier config file instead. So now there is no need to disable prettier/prettier. I have kept the rules of hooks fixes because they do fail on my local and I think it's the correct thing per my understanding. I have run "yarn lint" locally and it only returns warnings.

My mistake was I accidentally read the docs for @react-natve/eslint-plugin instead of eslint-config. That is my fault and I sincerely apologize. Let me know if there are any further questions.

**Review by trivikr — COMMENTED** [review]
LGTM

File: src/locale/i18n.web.ts [hunk 4]
```diff
@@ -110,12 +110,9 @@ export async function dynamicActivate(locale: AppLanguage) {
   i18n.activate(locale)
 }
 
-export async function useLocaleLanguage() {
+export function useLocaleLanguage() {
   const {appLanguage} = useLanguagePrefs()
   useEffect(() => {
-    const sanitizedLanguage = sanitizeAppLanguageSetting(appLanguage)
-
-    document.documentElement.lang = sanitizedLanguage
```
Comment (gaearon) [inline src/locale/i18n.web.ts:118]: i think we still need this, no?

Comment (PaulACoroneos) [reply src/locale/i18n.web.ts:118]: shame on me for copy pasting haha. Yes let me fix thus
